test-common/uk/gov/hmrc/economiccrimelevyregistration/EclTestData.scala

```diff
@@ -273,4 +276,23 @@ trait EclTestData {
     telephoneNumber = Some(alphaNumericString)
   )
 
+  implicit val arbRegistrationWithUnincorporatedAssociation: Arbitrary[RegistrationWithUnincorporatedAssociation] =
+    Arbitrary {
+      for {
+        registration          <- Arbitrary.arbitrary[Registration]
+        otherEntityJourneyData = OtherEntityJourneyData(
+                                   Some(UnincorporatedAssociation),
+                                   Some("UA Test"),
+                                   None,
+                                   None,
+                                   None,
+                                   Some("0123456789"),
+                                   None,
+                                   None,
+                                   Some("A1B YT4")
+                                 )
+      } yield RegistrationWithUnincorporatedAssociation(registration =
+        registration.copy(optOtherEntityJourneyData = Some(otherEntityJourneyData))
+      )
```

Comment:
should you also set entity type to 'Other' ?

Reply:
Fixed.